From 2a671f952c3125eb490d0468baf7dd3eb21c20f8 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@lucaweiss.eu>
Date: Wed, 11 Feb 2026 16:49:53 +0100
Subject: [PATCH] Include and link against libintl

textdomain, bindtextdomain and bind_textdomain_codeset are defined in
libintl.h, so include that header, and link the application against it.
---
 CMakeLists.txt              | 2 ++
 src/CMakeLists.txt          | 1 +
 src/gallery-application.cpp | 1 +
 3 files changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 55f463f6..2bcf643e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,6 +45,8 @@ find_package(PkgConfig REQUIRED)
 pkg_check_modules(EXIV2 REQUIRED exiv2)
 pkg_check_modules(MEDIAINFO REQUIRED libmediainfo)
 
+find_package(Intl REQUIRED)
+
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall")
 set(CMAKE_EXE_LINKER_FLAGS "-s")
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 53979732..67a0a521 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -67,6 +67,7 @@ target_link_libraries( ${GALLERY}
     Qt5::Quick
     Qt5::Sql
     Qt5::Widgets
+    Intl::Intl
     )
 
 install(TARGETS ${GALLERY} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
diff --git a/src/gallery-application.cpp b/src/gallery-application.cpp
index 12729203..8c409400 100644
--- a/src/gallery-application.cpp
+++ b/src/gallery-application.cpp
@@ -56,6 +56,7 @@
 #include <QTimer>
 #include <QUrl>
 #include <QtGui/QGuiApplication>
+#include <libintl.h>
 
 QElapsedTimer* GalleryApplication::m_timer = 0;
 
-- 
2.53.0

